source/extensions/filters/http/ip_tagging/ip_tagging_filter.cc

```diff
@@ -77,12 +77,19 @@ Http::FilterHeadersStatus IpTaggingFilter::decodeHeaders(Http::RequestHeaderMap&
     return Http::FilterHeadersStatus::Continue;
   }
 
+  if (!config_->customHeader().get().empty()) {
+    headers.remove(config_->customHeader());
+  }
```

Comment:
Is this necessary?

Reply:
> Is this necessary?

want to sanitize this header then the user can not spoof it; for example if I have a filter for internal ips; all internal traffic will have header `x-custom: internal`; but an external user can just set this header on the request, we want to avoid this case so need to remove x-custom header first. Is there a better way to do it?